lib/i18n_flat_yml_backend.rb

```diff
@@ -0,0 +1,77 @@
+# frozen_string_literal: true
+
+require 'i18n'
+require 'json'
+
+# Custom i18n backend that parse our "flat_yml" files into the nested
+# hash structure that i18n works with
+class I18nFlatYmlBackend < I18n::Backend::Simple
+  # @param filename [String] filename, assumed to have the locale slug in the filename ex "en.txt"
+  # @return [Array(Hash, Boolean)] tuple of a hash and keys_symbolized
+  def load_yml(filename)
+    content, keys_symbolized = super
+
+    if self.class.nested_hashes?(content)
+      [content, keys_symbolized]
+    else
+      [
+        {
+          self.class.locale(filename) => self.class.unflatten(content),
+        },
+        false,
+      ]
+    end
+  end
+
+  # @example
+  #   locale("config/locales/fr.yml")
+  #   # => "fr"
+  def self.locale(filename)
+    File.basename(filename, '.yml')
+  end
+
+  # @return [Boolean] true if +content+ appears to be a legacy "nested" yml file
+  #   instead of a flat yml file
+  def self.nested_hashes?(content)
+    content.is_a?(Hash) && content.keys.size == 1 && content[content.keys.first].is_a?(Hash)
+  end
```

Comment:
When are we expecting to deal with legacy files?

**Edit:** I see the `transliterate` and `telephony` files now, and your comment at https://github.com/18F/identity-idp/pull/10503#issuecomment-2078219671 . I guess I'd wonder if there's any reason we _couldn't_ update those to the new format? I think it'd be nice to be consistent and avoid the extra logic for handling different formats here.

Reply:
There are also yml files from other libraries that included that use the "traditional" structure

we could avoid this if we had our own separate extension?

Reply:
For the telephony.yml, that had a new strings with very specific whitespace handling so I felt it was easier in its own file?

Reply:
Yeah, I think we could keep this as-is for now